src/main/java/uk/gov/pay/webhooks/webhook/exception/WebhooksException.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.pay.webhooks.webhook.exception;
+
+public class WebhooksException extends RuntimeException {
+    protected static final String IDENTIFIER_GENERIC = "generic";
+    protected static final String IDENTIFIER_CALLBACK_URL_NOT_ON_ALLOW_LIST = "callback_url_not_on_allow_list";
+    protected static final String IDENTIFIER_CALLBACK_URL_MALFORMED = "callback_url_malformed";
+    protected static final String IDENTIFIER_CALLBACK_URL_PROTOCOL_NOT_SUPPORTED = "callback_url_protocol_not_supported";
+
+    private String errorIdentifier = IDENTIFIER_GENERIC;
```

Comment:
This should really be an enum. I know that would conflict with the `ErrorIdentifier` defined in Java commons but I think having an `ErrorIdentifierForCallbackUrlErrors` or `ErrorIdentifierForCallbackUrlErrorsIPinkySwearWeWillMergeThisWithTheMainErrorIdentifierSomeDay` would be preferable to using strings.

Reply:
It's only the admin tool that consumes the resources that can throw these exceptions at preset. 

I don't think we've got precedence for shared enums across frontend/ backend. 

Happy to help think it through but this enum currently couldn't be used by the consumer. I think it's a good idea to store them as one anyway to have one place to look in the Java app